### DISCUSSION TITLE
Restarting kernel when `python` is a shell wrapper that uses conda does not change binary path

### DISCUSSION THREAD

**Discussion by mvelbaum** [jupyterlab/jupyterlab#16759] [Q&A]
Hi,

My `python` binary is a shell script which activates a conda environment (which may change) and then runs the `python` binary within it. If I change the conda environment and restart the kernel jupyter seems to reuse the python binary from the previous one.
Is it caching this path anywhere and is there a way to override it?

Thanks!

**Comment by RRosio** [comment]
Hi @mvelbaum, maybe you can try taking a look at the [data files](https://docs.jupyter.org/en/latest/use/jupyter-directories.html#data-files), specifically the `kernels` subdirectory that contains the `kernel.json` file? Inspecting those contents might provide some more information about how the kernel is starting up. From there, maybe it could be possible to try using the path to the script as the Python binary?

**Comment by mvelbaum** [comment]
I ended up doing something else. I've just created a custom kernel that points to my python wrapper.
